### PR TITLE
feat(version): Add JSON/YAML output formats for version command

### LINKED ISSUE #513: Support machine-readable output for "kn version"
## Describe the feature:

In order to interoperate nicely with tooling that calls kn and in the spirit of our other commands we should support _machine-readable_ output in addition to _human-readable_ output for `kn version`.

I.e. I think having a `kn version -o json` return something like

```json
{
  "version": "v0.10.0",
  "date": "2019-11-21 21:06:36",
  "commit": "e07b5a9",
  "apis": {
     "serving": [
        "serving.knative.dev/v1alpha1"
     ],
     "eventing": [
        "sources.eventing.knative.dev/v1alpha1"
     ]
  } 
}
```

In the future there might be server version (i.e. which APIs are supported), included here as well.

### PR DESCRIPTION
**Pull request by navidshaikh**
Fixes #513

## Description:
Add -o flag to `kn version` command and support JSON/YAML formats.

## Changes
Examples:
```
➜  client git:(pr/kn-version-json) kn version -ojson
{
	"Version": "v20200309-local-1cbb94e2-dirty",
	"BuildDate": "2020-03-09 13:35:39",
	"GitRevision": "1cbb94e2",
	"SupportedAPIs": {
		"eventing": [
			"sources.eventing.knative.dev/v1alpha1 (knative-eventing v0.13.1)",
			"sources.eventing.knative.dev/v1alpha2 (knative-eventing v0.13.1)",
			"eventing.knative.dev/v1alpha1 (knative-eventing v0.13.1)"
		],
		"serving": [
			"serving.knative.dev/v1 (knative-serving v0.13.0)"
		]
	}
}
```
and
```
➜  client git:(pr/kn-version-json) kn version -oyaml
BuildDate: "2020-03-09 13:35:39"
GitRevision: 1cbb94e2
SupportedAPIs:
  eventing:
  - sources.eventing.knative.dev/v1alpha1 (knative-eventing v0.13.1)
  - sources.eventing.knative.dev/v1alpha2 (knative-eventing v0.13.1)
  - eventing.knative.dev/v1alpha1 (knative-eventing v0.13.1)
  serving:
  - serving.knative.dev/v1 (knative-serving v0.13.0)
Version: v20200309-local-1cbb94e2-dirty
```
/lint

### CODE REVIEW

**Comment by joshuawilson** [comment]
That looks good. What will be different when the version is a released version instead of a local build?

**Comment by navidshaikh** [comment]
@joshuawilson: 
If its a local build, we get: 
```
	"Version": "v20200309-local-1d4b497b",
```
while for released build:
```
	"Version": "v0.13.0",
```
rest of the fields should stay intact

**Comment by joshuawilson** [comment]
Thanks for fixing this. 
Do you want me to mark/tag anything?

**Comment by navidshaikh** [comment]
@joshuawilson thanks for the review!

/assign @rhuss

File: pkg/kn/commands/version/version_test.go [hunk 1]
```diff
@@ -16,31 +16,27 @@ package version
...
 Git Revision: {{.GitRevision}}
 Supported APIs:
 * Serving
-  - serving.knative.dev/v1 (knative-serving v0.13.0)
+  - {{ index .SupportedAPIs.serving 0}}
 * Eventing
-  - sources.eventing.knative.dev/v1alpha1 (knative-eventing v0.13.1)
-  - sources.eventing.knative.dev/v1alpha2 (knative-eventing v0.13.1)
-  - eventing.knative.dev/v1alpha1 (knative-eventing v0.13.1)
+  - {{ index .SupportedAPIs.eventing 0}}
```
Comment (rhuss) [inline pkg/kn/commands/version/version_test.go]: 👍 maybe we could use a gotemplate loop here ?

Comment (navidshaikh) [reply pkg/kn/commands/version/version_test.go]: done.

**Comment by navidshaikh** [comment]
@rhuss: addressed the review comment.

**Pull request merged**